codegenerator/cli/templates/static/codegen/src/db/DbFunctionsImplementation.js

```diff
@@ -417,16 +417,14 @@ module.exports.deleteStaleEndOfBlockRangeScannedDataForChain = (
     ;`;
 };
 
-module.exports.readDynamicContractsOnChainIdAtOrBeforeBlock = (
+module.exports.readDynamicContractsOnChainIdBeforeEventId = (
   sql,
   chainId,
-  block_number,
+  eventId,
 ) => sql`
-  SELECT c.contract_address, c.contract_type, c.event_id, c.chain_id, c.block_timestamp
-  FROM "public"."dynamic_contract_registry" as c
-  JOIN raw_events e ON c.chain_id = e.chain_id
-  AND c.event_id = e.event_id
-  WHERE e.block_number <= ${block_number} AND e.chain_id = ${chainId};`;
+  SELECT *
+  FROM "public"."dynamic_contract_registry"
+  WHERE event_id < ${eventId} AND chain_id = ${chainId};`;
```

Comment:
Now I construct the eventId of the next block and filter for anything below that ID so you end up with the desired effect